packages/api/lib/granules.js

```diff
@@ -62,10 +66,65 @@ function getGranuleProductVolume(granuleFiles = []) {
     .reduce((x, y) => x + y, 0);
 }
 
+/**
+ * Delete a Granule from Postgres and Dynamo, delete the Granule's
+ * Files from Postgres and S3
+ *
+ * @param {Object} params
+ * @param {Knex} params.knex - DB client
+ * @param {Object} params.dynamoGranule - Granule from DynamoDB
+ * @param {PostgresGranule} params.pgGranule - Granule from Postgres
+ * @param {FilePgModel} params.filePgModel - File Postgres model
+ * @param {GranulePgModel} params.granulePgModel - Granule Postgres model
+ * @param {Object} params.granuleModelClient - Granule Dynamo model
+ */
+const deleteGranuleAndFiles = async ({
+  knex,
+  dynamoGranule,
+  pgGranule,
+  filePgModel = new FilePgModel(),
+  granulePgModel = new GranulePgModel(),
+  granuleModelClient = new Granule(),
+}) => {
+  if (pgGranule === undefined) {
```

Comment:
While it makes logical sense to not attempt the PG deletes if the PG record does not in fact exist, there are also shouldn't be harm in having it do all the deletion logic, again because deletion is idempotent. Eventually the goal is to rip out the `get` calls from the granule DELETE endpoint altogether, so I'm somewhat inclined to not write this deletion handler with foreknowledge of whether a record exists

But we can leave it for now and deal with it if/when we refactor granule deletion to remove those .get calls

Reply:
Yeah, this will have to be re-evaluated when we remove the get calls. The reason we have this guard here is because we're using the `pgGranule`'s `cumulus_id` to fetch the PG Files. If we don't have the `pgGranule`, we know won't be able to get PG Files so we're skipping the attempt.